File: ford_fulkerson/test_ford_fulkerson.cpp
#include <iostream>
#include <cassert>
#include "ford_fulkerson.cpp"

void test_case_1() {
    FordFulkerson ff(6);
    ff.add_edge(0, 1, 16);
    ff.add_edge(0, 2, 13);
    ff.add_edge(1, 2, 10);
    ff.add_edge(1, 3, 12);
    ff.add_edge(2, 1, 4);
    ff.add_edge(2, 4, 14);
    ff.add_edge(3, 2, 9);
    ff.add_edge(3, 5, 20);
    ff.add_edge(4, 3, 7);
    ff.add_edge(4, 5, 4);

    assert(ff.max_flow(0, 5) == 23);
    cout << "Test case 1 passed!" << endl;
}

void test_case_2() {
    FordFulkerson ff(4);
    ff.add_edge(0, 1, 1000);
    ff.add_edge(1, 2, 1);
    ff.add_edge(2, 3, 1000);

    assert(ff.max_flow(0, 3) == 1);
    cout << "Test case 2 passed!" << endl;
}

void test_case_3() {
    FordFulkerson ff(2);
    ff.add_edge(0, 1, 10);

    assert(ff.max_flow(0, 1) == 10);
    cout << "Test case 3 passed!" << endl;
}

int main() {
    test_case_1();
    test_case_2();
    test_case_3();

    cout << "All test cases passed!" << endl;
    return 0;
}

File: Алгоритм Форда-Фалкерсона/test_ford_fulkerson.py
import unittest
from ford_fulkerson import FordFulkerson

class TestFordFulkerson(unittest.TestCase):
    def test_case_1(self):
        ff = FordFulkerson(6)
        edges = [
            (0, 1, 16),
            (0, 2, 13),
            (1, 2, 10),
            (1, 3, 12),
            (2, 1, 4),
            (2, 4, 14),
            (3, 2, 9),
            (3, 5, 20),
            (4, 3, 7),
            (4, 5, 4)
        ]
        for u, v, c in edges:
            ff.add_edge(u, v, c)
        self.assertEqual(ff.max_flow(0, 5), 23)

    def test_case_2(self):
        ff = FordFulkerson(4)
        edges = [
            (0, 1, 1000),
            (1, 2, 1),
            (2, 3, 1000)
        ]
        for u, v, c in edges:
            ff.add_edge(u, v, c)
        self.assertEqual(ff.max_flow(0, 3), 1)

    def test_case_3(self):
        ff = FordFulkerson(2)
        edges = [
            (0, 1, 10)
        ]
        for u, v, c in edges:
            ff.add_edge(u, v, c)
        self.assertEqual(ff.max_flow(0, 1), 10)

if __name__ == "__main__":
    unittest.main()
